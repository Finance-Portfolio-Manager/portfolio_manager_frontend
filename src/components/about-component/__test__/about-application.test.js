import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import AboutCard from './../about-application';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = shallow(<AboutCard></AboutCard>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<AboutCard></AboutCard>).toJSON();
    expect(tree).toMatchSnapshot();
})

test("renders title", ()=>{
    expect(wrapper.find("CardTitle").text()).toContain("About");
})

test("renders body/text", ()=>{
    expect(wrapper.find("CardText").length).toBe(1);
})