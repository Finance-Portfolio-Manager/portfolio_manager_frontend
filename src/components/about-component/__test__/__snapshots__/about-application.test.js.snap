// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="title border-0 card"
  id="about-card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      About
    </div>
    <p
      className="card-text"
    >
      Portfolio Manager was designed to be a central location for users to track their stock positions. Allows users to enter their stock purchases and sales, which will be reflected in our portfolio, showing stock specific and overall gains and losses. Leverages Yahoo Finance API to produce real-time stock details. Ensured application security by leveraging JSON Web Tokens and partial integration of Spring Security.
    </p>
  </div>
</div>
`;
